valhalla/sif/costfactory.h

```diff
@@ -53,41 +68,61 @@ template <class cost_t> class CostFactory {
       throw std::runtime_error("No costing provided to cost factory");
 
     // create the cost using the creation function
-    return Create(options.costing(), options);
+    auto costing_index = static_cast<int>(options.costing());
+    if (costing_index < options.costing_options_size()) {
+      return Create(options.costing_options(costing_index));
+    } // if we didnt have costing options we need to use some default ones
+    else {
+      throw std::runtime_error("No costing options provided to cost factory");
+    }
+  }
+
+  /**
+   * Make a default cost from its specified type
+   * @param costing  which costing to create
+   */
+  cost_ptr_t Create(Costing costing) const {
+    CostingOptions default_options;
+    default_options.set_costing(costing);
+    return Create(default_options);
   }
 
   /**
    * Make a cost from its specified type
    * @param costing  the type of cost to create
    * @param options  pbf with request options
    */
-  cost_ptr_t Create(const Costing costing, const Options& options) const {
-    auto itr = factory_funcs_.find(costing);
+  cost_ptr_t Create(const CostingOptions& options) const {
+    // you cant get a costing without a costing type
+    if (!options.has_costing())
+      throw std::runtime_error("No costing provided to cost factory");
+
+    auto itr = factory_funcs_.find(options.costing());
     if (itr == factory_funcs_.end()) {
-      auto costing_str = Costing_Enum_Name(costing);
+      auto costing_str = Costing_Enum_Name(options.costing());
       throw std::runtime_error("No costing method found for '" + costing_str + "'");
     }
     // create the cost using the function pointer
-    return itr->second(costing, options);
+    return itr->second(options);
   }
 
-  /**
-   * Convenience method to register all of the standard costing models.
-   */
-  void RegisterStandardCostingModels() {
-    Register(Costing::auto_, CreateAutoCost);
-    Register(Costing::auto_data_fix, CreateAutoDataFixCost);
-    Register(Costing::auto_shorter, CreateAutoShorterCost);
-    Register(Costing::bicycle, CreateBicycleCost);
-    Register(Costing::bus, CreateBusCost);
-    Register(Costing::hov, CreateHOVCost);
-    Register(Costing::taxi, CreateTaxiCost);
-    Register(Costing::motor_scooter, CreateMotorScooterCost);
-    Register(Costing::motorcycle, CreateMotorcycleCost);
-    Register(Costing::pedestrian, CreatePedestrianCost);
-    Register(Costing::truck, CreateTruckCost);
-    Register(Costing::transit, CreateTransitCost);
-    Register(Costing::none_, CreateNoCost);
+  mode_costing_t CreateModeCosting(const Options& options, TravelMode& mode) {
```

Comment:
this was happening a lot in different parts of the code so i pulled it into a function here